Build failure: forgejo

### Nixpkgs version

- Stable (25.11)

### Steps to reproduce

```command
❯ nix build github:NixOS/nixpkgs/nixos-25.11#forgejo
warning: error: unable to download 'https://cache.nixos.org/gpycp162xfif8xyg77q5rlg2aka8aala.narinfo': Resolving timed out after 5001 milliseconds (curl error code=28); retrying in 496ms ms (attempt 1/5)
error: builder for '/nix/store/36i8f271hs23zawphw17haj06rfxr8hg-forgejo-14.0.4.drv' failed with exit code 1;
       last 25 log lines:
       > 2026/04/14 09:46:10 ...s/storage/storage.go:157:initAttachments() [I] Initialising Attachment storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/attachments
       > 2026/04/14 09:46:10 ...s/storage/storage.go:147:initAvatars() [I] Initialising Avatar storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/avatars
       > 2026/04/14 09:46:10 ...s/storage/storage.go:173:initRepoAvatars() [I] Initialising Repository Avatar storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/repo-avatars
       > 2026/04/14 09:46:10 ...s/storage/storage.go:179:initRepoArchives() [I] Initialising Repository Archive storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/repo-archive
       > 2026/04/14 09:46:10 ...s/storage/storage.go:189:initPackages() [I] Initialising Packages storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/packages
       > 2026/04/14 09:46:10 ...s/storage/storage.go:200:initActions() [I] Initialising Actions storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/actions_log
       > 2026/04/14 09:46:10 ...s/storage/storage.go:204:initActions() [I] Initialising ActionsArtifacts storage with type: local
       > 2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/go-build1387888401/b001/data/actions_artifacts
       > 2026/04/14 09:46:10 ...er/issues/indexer.go:78:func1() [I] PID 51031: Initializing Issue Indexer: db
       > 2026/04/14 09:46:10 ...er/issues/indexer.go:78:func1() [I] PID 51031: Initializing Issue Indexer: bleve
       > --- FAIL: TestBleveDeleteIssue (5.01s)
       >     indexer_test.go:445:
       >               Error Trace:    /build/source/modules/indexer/issues/indexer_test.go:445
       >               Error:          Condition never satisfied
       >              Test:           TestBleveDeleteIssue
       >           Messages:       failed to delete issue
       > FAIL
       > FAIL      forgejo.org/modules/indexer/issues      5.181s
       > FAIL
       For full logs, run:
        nix log /nix/store/36i8f271hs23zawphw17haj06rfxr8hg-forgejo-14.0.4.drv
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/326407933

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/l7cn2f5was5a5y56a1jvf9lbkjc0rvz4-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/z37y57cp2n3fai1sjvr7221k41hprrf5-static-root-path.patch
patching file modules/setting/server.go
Hunk #1 succeeded at 338 (offset 21 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage ./.
Building subPackage ./contrib/environment-to-ini
buildPhase completed in 38 seconds
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
[33mhint: Using 'master' as the name for the initial branch. This default branch name[m
[33mhint: is subject to change. To configure the initial branch name to use in all[m
[33mhint: of your new repositories, which will suppress this warning, call:[m
[33mhint:[m
[33mhint: 	git config --global init.defaultBranch <name>[m
[33mhint:[m
[33mhint: Names commonly chosen instead of 'master' are 'main', 'trunk' and[m
[33mhint: 'development'. The just-created branch can be renamed via this command:[m
[33mhint:[m
[33mhint: 	git branch -m <name>[m
[33mhint:[m
[33mhint: Disable this message with "git config set advice.defaultBranchName false"[m
Initialized empty Git repository in /build/source/.git/
# cd /build/source; git -c log.showsignature=false log -1 --format=%H:%ct
fatal: your current branch 'master' does not have any commits yet
?   	forgejo.org	[no test files]
ok  	forgejo.org/build/codeformat	0.003s
ok  	forgejo.org/build/lint-locale	0.007s
?   	forgejo.org/build/lint-locale-usage	[no test files]
ok  	forgejo.org/build/lint-locale-usage/bin	0.034s
ok  	forgejo.org/cmd	0.809s
ok  	forgejo.org/cmd/forgejo	0.039s
?   	forgejo.org/contrib/environment-to-ini	[no test files]
ok  	forgejo.org/models	1.117s
ok  	forgejo.org/models/actions	6.353s
ok  	forgejo.org/models/activities	0.384s
?   	forgejo.org/models/admin	[no test files]
ok  	forgejo.org/models/asymkey	0.215s
ok  	forgejo.org/models/auth	0.552s
ok  	forgejo.org/models/avatars	0.133s
ok  	forgejo.org/models/db	0.876s
?   	forgejo.org/models/db/install	[no test files]
ok  	forgejo.org/models/db/paginator	0.133s
ok  	forgejo.org/models/dbfs	0.136s
ok  	forgejo.org/models/forgefed	0.020s
ok  	forgejo.org/models/forgejo/semver	0.131s
ok  	forgejo.org/models/git	0.276s
ok  	forgejo.org/models/issues	3.384s
?   	forgejo.org/models/moderation	[no test files]
ok  	forgejo.org/models/organization	0.787s
ok  	forgejo.org/models/packages	0.185s
?   	forgejo.org/models/packages/alpine	[no test files]
?   	forgejo.org/models/packages/alt	[no test files]
?   	forgejo.org/models/packages/conan	[no test files]
?   	forgejo.org/models/packages/conda	[no test files]
?   	forgejo.org/models/packages/container	[no test files]
?   	forgejo.org/models/packages/cran	[no test files]
ok  	forgejo.org/models/packages/debian	0.169s
?   	forgejo.org/models/packages/nuget	[no test files]
?   	forgejo.org/models/packages/rpm	[no test files]
?   	forgejo.org/models/perm	[no test files]
ok  	forgejo.org/models/perm/access	0.193s
ok  	forgejo.org/models/project	0.148s
?   	forgejo.org/models/pull	[no test files]
ok  	forgejo.org/models/quota	0.169s
ok  	forgejo.org/models/repo	0.806s
ok  	forgejo.org/models/secret	0.152s
?   	forgejo.org/models/shared/types	[no test files]
ok  	forgejo.org/models/system	0.131s
ok  	forgejo.org/models/unit	0.004s
?   	forgejo.org/models/unit/lint-locale-usage	[no test files]
ok  	forgejo.org/models/unittest	0.072s
ok  	forgejo.org/models/user	0.641s
ok  	forgejo.org/models/webhook	0.153s
ok  	forgejo.org/modules/actions	0.088s
ok  	forgejo.org/modules/activitypub	0.540s
ok  	forgejo.org/modules/analyze	0.019s
ok  	forgejo.org/modules/assetfs	0.005s
?   	forgejo.org/modules/auth	[no test files]
ok  	forgejo.org/modules/auth/openid	0.103s
?   	forgejo.org/modules/auth/pam	[no test files]
ok  	forgejo.org/modules/auth/password	0.005s
ok  	forgejo.org/modules/auth/password/hash	5.642s
ok  	forgejo.org/modules/auth/password/pwn	0.004s [no tests to run]
ok  	forgejo.org/modules/auth/webauthn	0.020s
ok  	forgejo.org/modules/avatar	0.119s
?   	forgejo.org/modules/avatar/identicon	[no test files]
ok  	forgejo.org/modules/base	0.021s
ok  	forgejo.org/modules/cache	0.173s
ok  	forgejo.org/modules/card	8.009s
ok  	forgejo.org/modules/charset	0.042s
?   	forgejo.org/modules/charset/ambiguous	[no test files]
?   	forgejo.org/modules/charset/invisible	[no test files]
ok  	forgejo.org/modules/container	0.003s
ok  	forgejo.org/modules/csv	0.029s
ok  	forgejo.org/modules/emoji	0.008s
ok  	forgejo.org/modules/eventsource	0.034s
ok  	forgejo.org/modules/forgefed	0.023s
ok  	forgejo.org/modules/generate	0.003s
ok  	forgejo.org/modules/git	0.934s
ok  	forgejo.org/modules/git/foreachref	0.003s
?   	forgejo.org/modules/git/internal	[no test files]
?   	forgejo.org/modules/git/pipeline	[no test files]
ok  	forgejo.org/modules/git/pushoptions	0.003s
ok  	forgejo.org/modules/git/url	0.003s
?   	forgejo.org/modules/gitrepo	[no test files]
ok  	forgejo.org/modules/graceful	0.004s
ok  	forgejo.org/modules/graceful/releasereopen	0.003s
ok  	forgejo.org/modules/hcaptcha	0.004s [no tests to run]
ok  	forgejo.org/modules/highlight	0.040s
ok  	forgejo.org/modules/hostmatcher	0.003s
?   	forgejo.org/modules/html	[no test files]
ok  	forgejo.org/modules/httpcache	0.004s
ok  	forgejo.org/modules/httplib	0.008s
ok  	forgejo.org/modules/indexer/code	0.327s
?   	forgejo.org/modules/indexer/code/bleve	[no test files]
ok  	forgejo.org/modules/indexer/code/bleve/tokenizer/hierarchy	0.003s
ok  	forgejo.org/modules/indexer/code/elasticsearch	0.029s
?   	forgejo.org/modules/indexer/code/internal	[no test files]
?   	forgejo.org/modules/indexer/internal	[no test files]
ok  	forgejo.org/modules/indexer/internal/bleve	0.007s
?   	forgejo.org/modules/indexer/internal/db	[no test files]
?   	forgejo.org/modules/indexer/internal/elasticsearch	[no test files]
?   	forgejo.org/modules/indexer/internal/meilisearch	[no test files]
2026/04/14 09:46:10 ...s/storage/storage.go:157:initAttachments() [I] Initialising Attachment storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/attachments
2026/04/14 09:46:10 ...s/storage/storage.go:147:initAvatars() [I] Initialising Avatar storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/avatars
2026/04/14 09:46:10 ...s/storage/storage.go:173:initRepoAvatars() [I] Initialising Repository Avatar storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/repo-avatars
2026/04/14 09:46:10 ...s/storage/storage.go:179:initRepoArchives() [I] Initialising Repository Archive storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/repo-archive
2026/04/14 09:46:10 ...s/storage/storage.go:189:initPackages() [I] Initialising Packages storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/packages
2026/04/14 09:46:10 ...s/storage/storage.go:200:initActions() [I] Initialising Actions storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/appdata3205684115/actions_log
2026/04/14 09:46:10 ...s/storage/storage.go:204:initActions() [I] Initialising ActionsArtifacts storage with type: local
2026/04/14 09:46:10 ...les/storage/local.go:33:NewLocalStorage() [I] Creating new Local Storage at /build/go-build1387888401/b001/data/actions_artifacts
2026/04/14 09:46:10 ...er/issues/indexer.go:78:func1() [I] PID 51031: Initializing Issue Indexer: db
2026/04/14 09:46:10 ...er/issues/indexer.go:78:func1() [I] PID 51031: Initializing Issue Indexer: bleve
--- FAIL: TestBleveDeleteIssue (5.01s)
    indexer_test.go:445: 
        	Error Trace:	/build/source/modules/indexer/issues/indexer_test.go:445
        	Error:      	Condition never satisfied
        	Test:       	TestBleveDeleteIssue
        	Messages:   	failed to delete issue
FAIL
FAIL	forgejo.org/modules/indexer/issues	5.181s
FAIL
```

### Additional context

Seems to be related to the backport of Forgejo 14.0.4 (#508695)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.78, NixOS, 25.11 (Xantusia), 25.11.20260401.bcd464c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.94.1
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/jalil/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/jalil/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/jalil/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/jalil/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/jalil/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/nnr3lkggmma4dkjavx1ibv04hiqq78zs-lix-2.94.1/share`
 - nixpkgs: `/nix/store/irldhwkzpmq4cqsp9mrf2an7gr7cbsir-source`



### Notify maintainers

@adamcstephens @bendlas @christoph-heiss @emilylange @nycodeghg @pyrox0 @tebriel

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fo/forgejo/generic.nix
+++ b/pkgs/by-name/fo/forgejo/generic.nix
@@ -12,6 +12,7 @@
   bash,
   brotli,
   buildGoModule,
+  fetchpatch,
   forgejo,
   git,
   gzip,
@@ -82,8 +83,15 @@ buildGoModule rec {
 
   patches = [
     ./static-root-path.patch
+  ]
+  ++ lib.optionals (lib.versionAtLeast version "14") [
+    # Backport fix for flaky TestBleveDeleteIssue test from v15.
+    # https://codeberg.org/forgejo/forgejo/pulls/11686
+    (fetchpatch {
+      url = "https://codeberg.org/forgejo/forgejo/commit/a32b0da87c10bb628a9d2203b700f0683e1ae966.patch";
+      hash = "sha256-mY1b35aKsiLNU56Ut/qiLrYj+IR8M2W8dZQiDqkDNxg=";
+    })
   ];
-
   postPatch = ''
     substituteInPlace modules/setting/server.go --subst-var data
   '';

